pkg/server/delta/v3/server.go

```diff
@@ -142,6 +142,14 @@ func (s *server) processDelta(str stream.DeltaStream, reqCh <-chan *discovery.De
 				return status.Errorf(codes.Unavailable, "empty request")
 			}
 
+			// make sure responses are processed prior to new requests to avoid deadlock
+			if len(watches.deltaMuxedResponses) > 0 {
+				go func() {
```

Comment:
This can potentially spin for a while (or forever) based on the internal of this select (which is undefined behavior). I think the implementation done in sotw (basically move reply in a separate method, and call here if needed) is more robust

Reply:
@valerian-roche do you mean purge the response channel before handling the request?

Reply:
Thought it again. This time I purge the response channel before processing a request, so we don't rely on the behavior of select which is undefined.